src/components/connect.js

```diff
@@ -17,9 +17,32 @@ function getDisplayName(WrappedComponent) {
 // Helps track hot reloading.
 let nextVersion = 0
 
-export default function connect(mapPropsToRequestsToProps, options = {}) {
+function connectFactory(options) {
+  function connectImpl(map) {
+    return connect(map, options)
+  }
+
+  connectImpl.withFetch = function (fetch) {
+    return connectFactory(assign({}, options, { fetch: fetch }))
```

Comment:
we might want to add some check here to make sure the passed argument at least is a function